tools/shared/src/main/scala/org/scalajs/core/tools/javascript/JSDesugaring.scala

```diff
@@ -160,6 +160,14 @@ private[javascript] object JSDesugaring {
 
     private val isStrongMode = outputMode == OutputMode.ECMAScript6StrongMode
 
+    // TODO Get rid of this when we break backward binary compatibility
+    private lazy val hasNewRuntimeLong = {
+      val rtLongClass = classEmitter.linkedClassByName(LongImpl.RuntimeLongClass)
+      rtLongClass.memberMethods.exists { linkedMethod =>
+        linkedMethod.tree.name.name == LongImpl.initFromParts
+      }
+    }
+
```

Comment:
Now this one I don't understand. You are not allowed to run a `0.6.x` linker with a `0.6.y` library if `y < x`.